src/accountsdb/db.zig

```diff
@@ -73,16 +102,14 @@ pub const AccountsDB = struct {
     /// This is tracked per-slot for purge/flush
     unrooted_accounts: RwMux(SlotPubkeyAccounts),
 
-    // pubkey->account LRU maps
+    /// pubkey->account LRU maps
     maybe_accounts_cache_rw: ?RwMux(AccountsCache),
 
     /// NOTE: see accountsdb/readme.md for more details on how these are used
-    file_map: RwMux(FileMap) = RwMux(FileMap).init(.{}),
+    file_map: RwMux(FileMap),
     /// `file_map_fd_rw` is used to ensure files in the file_map are not closed while its held as a read-lock.
     /// NOTE: see accountsdb/readme.md for more details on how these are used
-    file_map_fd_rw: std.Thread.RwLock = .{},
-
-    geyser_writer: ?*GeyserWriter,
+    file_map_fd_rw: std.Thread.RwLock,
```

Comment:
why remove helpful defaults?

Reply:
I think default fields are an anti-pattern in most cases. Here, all it does is save us a bit of typing in the `init` function, whilst introducing possible footguns if we were to ever update any logic or field relationships to no longer be proper when they are default-initialized. Requiring them to all be specified in the `init` function leaves less possibility for error.

I think the main valid cases for default field values is:
* Pure data fields: each field can be initialized to any value independently of the others.
* API backwards-compatibility: an API allowed direct-initialization of a struct, and it needs to add fields in a backwards-compatible way.

Also see: https://ziglang.org/documentation/master/#Faulty-Default-Field-Values